engine/pkg/meta/internal/sqlkv/connection.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqlkv
+
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+	"sync"
+
+	"go.uber.org/atomic"
+
+	"github.com/pingcap/tiflow/engine/pkg/dbutil"
+	"github.com/pingcap/tiflow/engine/pkg/meta/model"
+	"github.com/pingcap/tiflow/pkg/errors"
+)
+
+// NewClientConnImpl return a new clientConnImpl
+func NewClientConnImpl() *clientConnImpl {
+	return &clientConnImpl{
+		isInitialized: atomic.NewBool(false),
+	}
+}
+
+type clientConnImpl struct {
+	mu            sync.Mutex
+	isInitialized *atomic.Bool
+	db            *sql.DB
+}
+
+// Initialize implements Initialize of ClientConn
+func (cc *clientConnImpl) Initialize(storeConf *model.StoreConfig) error {
+	if storeConf == nil {
+		return errors.ErrMetaParamsInvalid.GenWithStackByArgs("storeConf is nil")
+	}
+
+	if storeConf.StoreType != model.StoreTypeSQL {
```

Comment:
How does the caller know that `storeConf.StoreType` is expected to be `model.StoreTypeSQL`?

Reply:
I seems a little wierd here. I will move the `storeConf *model.StoreConf` to constructor.